modules/services/nix-gc.nix

```diff
@@ -0,0 +1,115 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.nix.gc;
+
+  mkCalendarInterval = frequency:
+    let
+      freq = {
+        "hourly" = [{ Minute = 0; }];
+        "weekly" = [{
+          Weekday = 1;
+          Hour = 0;
+          Minute = 0;
+        }];
+        "monthly" = [{
+          Day = 1;
+          Hour = 0;
+          Minute = 0;
+        }];
+        "semiannually" = [
+          {
+            Month = 1;
+            Day = 1;
+            Hour = 0;
+            Minute = 0;
+          }
+          {
+            Month = 7;
+            Day = 1;
+            Hour = 0;
+            Minute = 0;
+          }
+        ];
+        "annually" = [{
+          Month = 1;
+          Day = 1;
+          Hour = 0;
+          Minute = 0;
+        }];
+      };
+    in freq.${frequency};
+  nix-package = if config.nix.enable && config.nix.package != null then
+    config.nix.package
+  else
+    pkgs.nix;
+in {
+  meta.maintainers = [ maintainers.shivaraj-bh ];
+
+  options = {
+    nix.gc = {
+      automatic = mkOption {
+        default = false;
+        type = types.bool;
+        description = lib.mdDoc
```

Comment:
The `lib.mdDoc` function is unnecessary. The description is interpreted as Markdown by default.